src/containers/navigation-icons/AccountIcon.tsx

```diff
@@ -0,0 +1,60 @@
+import { useCallback, FC } from 'react'
+import { useTranslation } from 'react-i18next'
+import { useAppSelector } from '~/hooks/use-redux'
+import { AxiosResponse } from 'axios'
+
+import Avatar from '@mui/material/Avatar'
+import Tooltip from '@mui/material/Tooltip'
+
+import { userService } from '~/services/user-service'
+import useAxios from '~/hooks/use-axios'
+import { defaultResponses } from '~/constants'
+
+import { styles } from '~/containers/navigation-icons/NavigationIcons.styles'
+
+import { UserResponse, UserRole } from '~/types'
+
+interface AccountIconProps {
+  openMenu: () => void
+}
+
+const AccountIcon: FC<AccountIconProps> = ({ openMenu }) => {
+  const { t } = useTranslation()
+  const { userId, userRole } = useAppSelector((state) => state.appMain)
+
+  const getUserData: () => Promise<AxiosResponse<UserResponse>> = useCallback(
+    () => userService.getUserById(userId, userRole as UserRole),
+    [userId, userRole]
+  )
+
+  const { loading, response } = useAxios<UserResponse>({
```

Comment:
```suggestion
  const { loading, response: { photo, firstName, LastName } } = useAxios<UserResponse>({
```

And then you can use them directly + in lines 42-43, you can make them more readable by writting
`firstName[0] || ''` && `lastName[0] || ''`
What do you think about this approach?

Reply:
Thank you for this advice! Realy looks much better and more readable)